libtenzir/builtins/operators/deduplicate.cpp

```diff
@@ -179,18 +204,68 @@ class plugin final : public operator_plugin2<deduplicate_operator> {
 public:
   auto make(invocation inv, session ctx) const
     -> failure_or<operator_ptr> override {
-    auto key = std::optional<ast::expression>{};
+    auto selector_exprs = std::vector<ast::expression>{};
+    auto expression_key = std::optional<ast::expression>{};
```

Comment:
I think we can just get rid of this artificial distinction and always work on a single `expressions` vector:

```suggestion
    auto expressions = std::vector<ast::expression>{};
```

This will also simplify both the over `inv.args` below.